lib/travis/api/v3/renderer/model_renderer.rb

```diff
@@ -33,8 +33,8 @@ def initialize(model, script_name: nil, **options)
 
     def href
       return @href if defined? @href # allows setting href to nil
-      return unless self.class.type and model.respond_to? :id and model.id
-      @href = Renderer.href(self.class.type, script_name: script_name, id: model.id)
+      return unless self.class.type and model.respond_to? :attributes
+      @href = Renderer.href(self.class.type, model.attributes, script_name: script_name)
```

Comment:
This change is to allow links generated from other attributes than id (like branch.name). See above.